contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,76 @@
+
+### <a name="cache"></a>Cache
+Data within the messages for communications between the server and the client should be indicated if it is cacheable. If cacheable, the caches at the elements along the line of communication (e.g. the client cache, the server cache, the proxy cache etc.) will store the data and reuse if for identical requests later. The cacheability of information reduces the amount of interactions needed between the client and the server to access information, and thus improves the efficiency of the system. The average latency of interactions is also reduced, which leads to faster response to the client and an improved user-perceived performance.
+
+### <a name="layeredSystem"></a>Layered System
+There should be multiple layers between client and server. The layers are made up by various [intermediaries](https://www.techopedia.com/definition/24378/web-intermediary-wbi) between the client and the server facilitating the processing tasks. Some examples of intermediaries are load-balancer, cache etc. The system elements have no knowledge of the things outside their own layers. For example, a client would not know if it is connected directly to the server or to an intermediary, and an intermediary would not know if it is connected to another intermediary. By limiting the scope of the system element, the complexity of the system is greatly reduced. The system becomes more modifiable as there are less dependencies between the system elements. By facilitating the processing tasks, intermediaries can enhance the server performance by reducing the redundant server processing. Moreover, as the intermediaries can carry out a wide range of tasks including [encryption](https://en.wikipedia.org/wiki/Encryption) and request modification and etc.(see "MEGs" in [this article](http://www.almaden.ibm.com/cs/wbi/doc/Architecture.html)), the organizations can use intermediaries to enhance their security and their control over the information. 
```

Comment:
Is it the server facilitating the processing tasks or the intermediaries? What are you currently implying in your paragraph here?

Reply:
"...intermediaries can enhance the server performance by reducing ~the~ redundant server processing..."

Reply:
"...and their control over ~the~ information."